addons/mail/static/src/models/thread/thread.js

```diff
@@ -419,9 +419,13 @@ function factory(dependencies) {
                 orderBy: [{ name: 'id', asc: false }],
             }));
             for (const attachmentData of attachmentsData) {
-                this.env.models['mail.attachment'].insert(Object.assign({
-                    originThread: [['link', this]],
-                }, this.env.models['mail.attachment'].convertData(attachmentData)));
+                this.update({
+                    originThreadAttachments: [['insert-and-replace',
+                       attachmentsData.map(data =>
+                          this.env.models['mail.attachment'].convertData(data)
+                       )
+                    ]],
+                });
             }
```

Comment:
Oops, also this `for of` loop should be removed, because `attachmentsData` is used in the command. `attachmentData` is unused as a result.